src/java/org/apache/cassandra/db/commitlog/CommitLog.java

```diff
@@ -122,6 +135,9 @@ private static CommitLog construct()
 
         // register metrics
         metrics.attach(executor, segmentManager);
+
+        ConfigurationRegistry.instance.addPropertyChangeListener(CDC_BLOCK_WRITES, this::cdcBlockWritesChangeListener, Boolean.class);
```

Comment:
Looks like the majority of complexity of this patch is to add listeners for this single listener?  There maybe more cases as you flesh this out, but at least in the current PR this is the only real listener (the others just log you mutated, so don't need to listen).

I feel we could simplify to

```ConfigurationRegistry.instance.register("cdc_block_writes", this:: setCDCBlockWrites0)```

Where `setCDCBlockWrites0` is just the logic in `cdcBlockWritesChangeListener`

I don't like the `Boolean.class` as it's brittle and can drift; in fact this patch provides the wrong type!  You want `Boolean.TYPE`.